openjdk-integ-tests/src/test/java/org/conscrypt/javax/crypto/CipherTest.java

```diff
@@ -0,0 +1,4766 @@
+        PrintStream out = new PrintStream(errBuffer);
+
+        Set<String> seenBaseCipherNames = new HashSet<String>();
+        Set<String> seenCiphersWithModeAndPadding = new HashSet<String>();
+
+        Provider[] providers = Security.getProviders();
+        for (Provider provider : providers) {
+            Set<Provider.Service> services = provider.getServices();
+            for (Provider.Service service : services) {
+                String type = service.getType();
+                if (!type.equals("Cipher")) {
+                    continue;
+                }
+
+                String algorithm = service.getAlgorithm().toUpperCase(Locale.US);
+
+                /*
+                 * Any specific modes and paddings aren't tested directly here,
+                 * but we need to make sure we see the bare algorithm from some
+                 * provider. We will test each mode specifically when we get the
+                 * base cipher.
+                 */
+                final int firstSlash = algorithm.indexOf('/');
+                if (firstSlash == -1) {
+                    seenBaseCipherNames.add(algorithm);
+                } else {
+                    final String baseCipherName = algorithm.substring(0, firstSlash);
+                    if (!seenBaseCipherNames.contains(baseCipherName)
+                            && !(baseCipherName.equals("AES_128")
+                                || baseCipherName.equals("AES_192")
+                                || baseCipherName.equals("AES_256"))) {
+                        seenCiphersWithModeAndPadding.add(baseCipherName);
+                    }
+                    if (!Conscrypt.isConscrypt(provider)) {
+                        continue;
+                    }
+                }
+
+                if (provider.getName().equals("SunJCE")) {
+                    continue;
+                }
+
+                try {
+                    test_Cipher_Algorithm(provider, algorithm);
+                } catch (Throwable e) {
+                    out.append("Error encountered checking " + algorithm
+                               + " with provider " + provider.getName() + "\n");
+                    e.printStackTrace(out);
+                }
+
+                Set<String> modes = StandardNames.getModesForCipher(algorithm);
+                if (modes != null) {
+                    for (String mode : modes) {
+                        Set<String> paddings = StandardNames.getPaddingsForCipher(algorithm);
+                        if (paddings != null) {
+                            for (String padding : paddings) {
+                                final String algorithmName = algorithm + "/" + mode + "/" + padding;
+                                try {
+                                    test_Cipher_Algorithm(provider, algorithmName);
+                                } catch (Throwable e) {
+                                    out.append("Error encountered checking " + algorithmName
```

Comment:
This is really strange ...  We gather all errors and then throw at the end of the method with a mongo exception. Breaking this into multiple tests is a far better solution, but it will take some work.  You should definitely follow this up with a cleanup PR.